### ISSUE TITLE
Unit tests failing/long running in 0.21 version

### ISSUE THREAD

**Issue by maddanki95** [quickwit-inc/tantivy#2202]
This [commit](https://github.com/quickwit-oss/tantivy/pull/2094) is causing following failures:

running 5 tests
test test_fail_on_commit_segment has been running for over 60 seconds
test test_fail_on_flush_segment has been running for over 60 seconds
test test_fail_on_flush_segment_but_one_worker_remains has been running for over 60 seconds
test test_failpoints_managed_directory_gc_if_delete_fails has been running for over 60 seconds
test test_write_commit_fails has been running for over 60 seconds
test test_fail_on_flush_segment_but_one_worker_remains ... FAILED
test test_write_commit_fails ... FAILED
test test_failpoints_managed_directory_gc_if_delete_fails ... FAILED
test test_fail_on_commit_segment ... FAILED
test test_fail_on_flush_segment ... FAILED

failures:

---- test_fail_on_flush_segment_but_one_worker_remains stdout ----
thread 'test_fail_on_flush_segment_but_one_worker_remains' panicked at 'add_document should have returned an error', tests/failpoints/mod.rs:107:5
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

---- test_write_commit_fails stdout ----
thread 'test_write_commit_fails' panicked at 'assertion failed: index_writer.commit().is_err()', tests/failpoints/mod.rs:57:5

---- test_failpoints_managed_directory_gc_if_delete_fails stdout ----
thread 'test_failpoints_managed_directory_gc_if_delete_fails' panicked at 'assertion failed: managed_directory.exists(test_path).unwrap()', tests/failpoints/mod.rs:31:5

---- test_fail_on_commit_segment stdout ----
thread 'test_fail_on_commit_segment' panicked at 'assertion failed: index_writer.commit().is_err()', tests/failpoints/mod.rs:122:5

---- test_fail_on_flush_segment stdout ----
thread 'test_fail_on_flush_segment' panicked at 'add_document should have returned an error', tests/failpoints/mod.rs:88:5

failures:
    test_fail_on_commit_segment
    test_fail_on_flush_segment
    test_fail_on_flush_segment_but_one_worker_remains
    test_failpoints_managed_directory_gc_if_delete_fails
    test_write_commit_fails

test result: FAILED. 0 passed; 5 failed; 0 ignored; 0 measured; 0 filtered out; finished in 2892.20s

**Steps:**

1. Cloned the git repo
2. checked out to 0.21 tag
```
commit 49448b31c69ceec4cbab0f68c38f608b0dbd1a3b (HEAD, tag: 0.21)
Author: PSeitz <PSeitz@users.noreply.github.com>
Date:   Fri Sep 1 13:58:58 2023 +0200

    chore: Release (#2168)
```    
3. cargo build
4. cargo test

**Note:**
Also tests::test_validate_checksum  is long running now in 0.21 version

**Comment by PSeitz** [comment]
Both issues have been fixed on main

**Comment by maddanki95** [comment]
Can you please provide the commits for those fixes namely  failpoints test issue and then the test_validate_checksum issue?

**Comment by PSeitz** [comment]
They are both in one PR: https://github.com/quickwit-oss/tantivy/pull/2176

**Comment by PSeitz** [comment]
I'll close this as this is fixed

**Comment by PSeitz** [comment]
The fix has been released with `0.21.1`
